src/nfd_win.cpp

```diff
@@ -360,14 +367,15 @@ nfdresult_t NFD_OpenDialog( const char *filterList,
     HRESULT result = ::CoInitializeEx(NULL,
                                       ::COINIT_APARTMENTTHREADED |
                                       ::COINIT_DISABLE_OLE1DDE );
+                                      
+    ::IFileOpenDialog *fileOpenDialog(NULL);
```

Comment:
GCC considers it an error when a goto jump crosses variable initialisation, thus fileOpenDialog has to be initialized a bit earlier.